src/marten/db/query/set.cr

```diff
@@ -325,6 +325,40 @@ module Marten
           add_query_node(query_node)
         end
 
+        # Returns if a query set includes a specific model.
+        def includes?(value : DB::Model)
```

Comment:
Another thing to take into account is that we probably want to check that the specified record is persisted. If that's not the case and if the record does not have a primary key yet, we could raise a `Marten::DB::Errors::UnmetQuerySetCondition` exception explaining the situation.

What do you think?

Reply:
There is a [`#persisted?`](https://martenframework.com/docs/api/0.2/Marten/DB/Model/Persistence.html#persisted%3F-instance-method) method available on all model record objects that you can use!